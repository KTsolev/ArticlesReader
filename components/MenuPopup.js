	
import React, { Component } from 'react';
import { View, Text, Image, TouchableOpacity  } from 'react-native';
import { Icon } from 'react-native-elements';
import { connect } from 'react-redux';
import { showHidePopupMenu } from '../actions/actions';

class MenuPopup extends Component {
  constructor(props) {
      super(props);
      this.state = {
          showPopUpMenu: props.showPopUpMenu || false,
      };
  }

  componentDidUpdate(prevProps) {
    if(prevProps.showPopUpMenu !== this.props.showPopUpMenu) {
        this.setState({showPopUpMenu: this.props.showPopUpMenu});
    }
  }Â 


  clickHandler = () => {
    this.props.togglePopup(!this.state.showPopUpMenu);
    this.props.navigation.navigate('Home');
  };

  render() {
    return this.state.showPopUpMenu ? <View style={{ position: 'absolute', top: '100%', right: 0, width: 120, height: 40, borderWidth: 1, borderColor: '#a8a8a8', backgroundColor: '#ededed' }}>
          <TouchableOpacity style={{ flexDirection: 'row', alignItems: 'center', justifyContent: 'space-around', padding: 5, width: '100%', height: '100%' }} onPress={this.clickHandler}>
            <Icon name='home' size={16} type='font-awesome' color='#5b5e61' />
            <Text>Home Page</Text>
          </TouchableOpacity>
      </View> : null;
  }
}

const mapStateToProps = (state)=> {
    console.log(state);
    return {
        showPopUpMenu: state.uiElementsState.showPopUpMenu
    }
  }
  
  const mapDispatchToProps = dispatch => {
    return {
      togglePopup:(toggle) => {
        dispatch(showHidePopupMenu(toggle))
      }
    }
  }
  
 
  export default connect(mapStateToProps, mapDispatchToProps)(MenuPopup);