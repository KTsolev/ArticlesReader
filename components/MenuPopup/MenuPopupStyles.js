import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: { position: 'absolute', top: '100%', right: 0, width: 120,
        height: 40, borderWidth: 1, borderColor: '#a8a8a8', backgroundColor: '#ededed' },
    button: { flexDirection: 'row', alignItems: 'center', justifyContent: 'space-around',
        padding: 5, width: '100%', height: '100%' } 
});