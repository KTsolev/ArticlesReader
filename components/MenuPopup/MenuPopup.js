import React, { Component } from 'react';
import { View, Text, TouchableOpacity  } from 'react-native';
import { Icon } from 'react-native-elements';
import { connect } from 'react-redux';
import { showHidePopupMenu } from '../../actions/actions';
import { styles } from './MenuPopupStyles';

class MenuPopup extends Component {
  constructor(props) {
      super(props);
      this.state = {
          showPopUpMenu: props.showPopUpMenu || false,
      };
  }

  componentDidUpdate(prevProps) {
    if(prevProps.showPopUpMenu !== this.props.showPopUpMenu) {
        this.setState({showPopUpMenu: this.props.showPopUpMenu});
    }
  }Â 


  clickHandler = () => {
    this.props.togglePopup(!this.state.showPopUpMenu);
    this.props.navigation.navigate('Home');
  };

  render() {
    const { container, button } = styles;
    return this.state.showPopUpMenu ? <View style={container}>
          <TouchableOpacity style={button} onPress={this.clickHandler}>
            <Icon name='home' size={16} type='font-awesome' color='#5b5e61' />
            <Text>Home Page</Text>
          </TouchableOpacity>
      </View> : null;
  }
}

const mapStateToProps = (state)=> {
    return {
        showPopUpMenu: state.uiElementsState.showPopUpMenu
    }
  }
  
  const mapDispatchToProps = dispatch => {
    return {
      togglePopup:(toggle) => {
        dispatch(showHidePopupMenu(toggle))
      }
    }
  }
  
 
  export default connect(mapStateToProps, mapDispatchToProps)(MenuPopup);