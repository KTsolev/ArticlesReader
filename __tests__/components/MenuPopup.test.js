import MenuPopup from '../../components/MenuPopup/MenuPopup';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';

const createTestProps = (props) => ({
    navigation: {
      navigate: jest.fn(),
      getParam: (name) => {
          if (name==='url') {
            return 'https://jestjs.io/';
          } 
          return null;
      }
    },
    ...props
  });

describe('<MenuPopup />', () => {
    let props;
    let middlewares;
    let mockStore;
    const componentWillUnmount = jest.fn();
    const pressGestureHandler = jest.fn();

    beforeEach(() => {
        props = createTestProps({});
        middlewares = [thunk];
        mockStore = configureMockStore(middlewares);
        props.store = mockStore({
            uiElementsState: {
                showFilters: false,
                showPopUpMenu: false
            },
            articles: {
                filters: {}
            },
            togglePopup: jest.fn(),
        });
    });
    
    test('renders the component', () => {
        const wrapper = shallow(<MenuPopup {...props}/>);
        const component = wrapper.dive();
        expect(toJson(component)).toMatchSnapshot();
    });

});