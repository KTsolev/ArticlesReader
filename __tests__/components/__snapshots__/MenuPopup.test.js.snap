// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuPopup /> renders the component 1`] = `
<MenuPopup
  navigation={
    Object {
      "getParam": [Function],
      "navigate": [MockFunction],
    }
  }
  showPopUpMenu={false}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  togglePopup={[Function]}
/>
`;
